string[] nodes_init
string[] nodes_ready
string[] nodes_error